### PR TITLE
docs: add section about passing in JWTs

### PR DESCRIPTION
**Pull request by mjfwebb**

### CODE REVIEW

**Comment by neo-technology-commit-status-publisher** [comment]
Thanks for the documentation updates.

The preview documentation has now been torn down - reopening this PR will republish it.

File: modules/ROOT/pages/authentication-and-authorization/configuration.adoc [hunk 1]
```diff
@@ -126,3 +126,50 @@ type JWT @jwt {
 ====
 The seemingly excessive escaping is required to doubly escape: once for GraphQL and once for `dot-prop`, which is used under the hood to resolve the path.
 ====
+
+== Passing in JWTs
+
+If you wish to pass in an encoded JWT, this must go into the token field of the context.
```
Comment (lidiazuin) [inline modules/ROOT/pages/authentication-and-authorization/configuration.adoc]:
```suggestion
To pass in an encoded JWT, you must use the token field of the context.
```

File: modules/ROOT/pages/authentication-and-authorization/configuration.adoc [hunk 2]
```diff
@@ -126,3 +126,50 @@ type JWT @jwt {
...
+
```
Comment (lidiazuin) [inline modules/ROOT/pages/authentication-and-authorization/configuration.adoc]:
```suggestion
```

File: modules/ROOT/pages/authentication-and-authorization/configuration.adoc [hunk 3]
```diff
@@ -126,3 +126,50 @@ type JWT @jwt {
...
+Using Apollo Server as an example, you extract the authorization header into the token property of the context as follows:
```
Comment (lidiazuin) [inline modules/ROOT/pages/authentication-and-authorization/configuration.adoc]:
```suggestion
When using Apollo Server, extract the authorization header into the token property of the context as follows:
```

File: modules/ROOT/pages/authentication-and-authorization/configuration.adoc [hunk 4]
```diff
@@ -126,3 +126,50 @@ type JWT @jwt {
...
+const server = new ApolloServer({
+    schema,
+});
+
+await startStandaloneServer(server, {
+    context: async ({ req }) => ({ token: req.headers.authorization }),
+});
+----
+
+For example, a HTTP request with the following `authorization` header:
```
Comment (lidiazuin) [inline modules/ROOT/pages/authentication-and-authorization/configuration.adoc]:
```suggestion
For example, a HTTP request with the following `authorization` header should look like this:
```

File: modules/ROOT/pages/authentication-and-authorization/configuration.adoc [hunk 5]
```diff
@@ -126,3 +126,50 @@ type JWT @jwt {
...
+[source]
+----
+POST / HTTP/1.1
+authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyLCJyb2xlcyI6WyJ1c2VyX2FkbWluIiwicG9zdF9hZG1pbiIsImdyb3VwX2FkbWluIl19.IY0LWqgHcjEtOsOw60mqKazhuRFKroSXFQkpCtWpgQI
+content-type: application/json
+----
+
+=== Decoded JWTs
+
+Alternatively, you can pass a key `jwt` of type `JwtPayload` into the context, which has the following definition:
```
Comment (lidiazuin) [inline modules/ROOT/pages/authentication-and-authorization/configuration.adoc:155]:
```suggestion
Alternatively, you can pass a key `jwt` of type `JwtPayload` into the context:
```

File: modules/ROOT/pages/authentication-and-authorization/configuration.adoc [hunk 6]
```diff
@@ -126,3 +126,50 @@ type JWT @jwt {
...
+    sub?: string | undefined;
+    aud?: string | string[] | undefined;
+    exp?: number | undefined;
+    nbf?: number | undefined;
+    iat?: number | undefined;
+    jti?: string | undefined;
+}
+----
+
+_Do not_ pass in the header or the signature.
```
Comment (lidiazuin) [inline modules/ROOT/pages/authentication-and-authorization/configuration.adoc:173]:
```suggestion
[WARNING]
_Do not_ pass in the header or the signature.
```

File: modules/ROOT/pages/authentication-and-authorization/configuration.adoc [hunk 7]
```diff
@@ -126,3 +126,50 @@ type JWT @jwt {
...
+For example, a HTTP request with the following `authorization` header:
+
+[source]
+----
+POST / HTTP/1.1
+authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyLCJyb2xlcyI6WyJ1c2VyX2FkbWluIiwicG9zdF9hZG1pbiIsImdyb3VwX2FkbWluIl19.IY0LWqgHcjEtOsOw60mqKazhuRFKroSXFQkpCtWpgQI
+content-type: application/json
+----
+
+=== Decoded JWTs
```
Comment (lidiazuin) [inline modules/ROOT/pages/authentication-and-authorization/configuration.adoc]:
```suggestion
```

**Review by lidiazuin — COMMENTED** [review]
Thanks Michael! I left some editorial suggestions.

File: modules/ROOT/pages/authentication-and-authorization/configuration.adoc (hunk 5 above)
Comment (mjfwebb) [reply modules/ROOT/pages/authentication-and-authorization/configuration.adoc:155]: Hmm this one might need some reworking. The code this refers to is actually not an example of how to use the payload, it is instead the shape of the payload.

We didn't want to go into too much detail about what a user would need to do specifically, instead just providing them enough information to make informed decisions.

Comment (lidiazuin) [reply modules/ROOT/pages/authentication-and-authorization/configuration.adoc:155]: I see! It's ok, so just ignore my suggestion :)

File: modules/ROOT/pages/authentication-and-authorization/configuration.adoc [hunk 8]
```diff
@@ -126,3 +126,50 @@ type JWT @jwt {
...
+
+For example, a HTTP request with the following `authorization` header should look like this:
+
+[source]
+----
+POST / HTTP/1.1
+authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyLCJyb2xlcyI6WyJ1c2VyX2FkbWluIiwicG9zdF9hZG1pbiIsImdyb3VwX2FkbWluIl19.IY0LWqgHcjEtOsOw60mqKazhuRFKroSXFQkpCtWpgQI
+content-type: application/json
+----
+
```
Comment (lidiazuin) [inline modules/ROOT/pages/authentication-and-authorization/configuration.adoc:154]:
```suggestion
```

there's an extra space here :)

File: modules/ROOT/pages/authentication-and-authorization/configuration.adoc [hunk 9]
```diff
@@ -126,3 +126,50 @@ type JWT @jwt {
...
+    sub?: string | undefined;
+    aud?: string | string[] | undefined;
+    exp?: number | undefined;
+    nbf?: number | undefined;
+    iat?: number | undefined;
+    jti?: string | undefined;
+}
+----
+
+[WARNING]
```
Comment (lidiazuin) [inline modules/ROOT/pages/authentication-and-authorization/configuration.adoc:172]:
```suggestion
```

duplicated

**Review by lidiazuin — APPROVED** [review]

File: modules/ROOT/pages/authentication-and-authorization/configuration.adoc (hunk 9 above)
Comment (mjfwebb) [reply modules/ROOT/pages/authentication-and-authorization/configuration.adoc:172]: haha whoops

**Pull request merged**